pkg/cmd/server/server.go

```diff
@@ -728,9 +728,9 @@ func (s *server) runControllers(
 	restorer, err := restore.NewKubernetesRestorer(
 		s.discoveryHelper,
 		client.NewDynamicFactory(s.dynamicClient),
-		nil,
 		s.config.restoreResourcePriorities,
 		s.arkClient.ArkV1(),
+		s.sharedInformerFactory.Ark().V1().VolumeSnapshotLocations(),
```

Comment:
we need to make sure that this has synced before using it.  This is always a little trickier when a non-controller takes a dependency on a lister/informer.  One option would be to have the restore controller wait for it to sync, then pass it as an argument to the `Restore` func.  Or, you could pass it to both the restore controller and the restorer, but have the restore controller wait for it to sync, and have the restorer's `Restore` func just return an error if it hasn't synced yet.  Since logically we'll only call `Restore` once all of the restore controller's informers have synced, we should be good.

Reply:
Yeah, I think I'll just pass it to the controller, wait for the sync, and pass it in on the `Restore` call.